main/pipeline/semantic_extension/SemanticExtension.h

```diff
@@ -26,7 +27,7 @@ namespace pipeline::semantic_extension {
 class SemanticExtension {
 public:
     virtual void typecheck(const core::GlobalState &, cfg::CFG &, std::unique_ptr<ast::MethodDef> &) const = 0;
-    virtual void patchDSL(const core::GlobalState &, ast::ClassDef *) const = 0;
+    virtual void patchDSL(core::MutableContext &, ast::ClassDef *) const = 0;
```

Comment:
this code is run in parallel in global states that are thrown away.
Also, as symbols haven't been registered yet, `Context.owner` is always `<root>` or something nonsensical.
What are you trying to do?